From: Werner Koch <wk@gnupg.org>
Date: Mon, 3 Jun 2019 16:31:58 +0200
Subject: Return better error code for some getinfo IPC commands.

* agent/command.c (cmd_getinfo): Return GPG_ERR_FALSE as boolean False.
* g13/server.c (cmd_getinfo): Ditto.
* sm/server.c (cmd_getinfo): Ditto.
--

GPG_ERR_FALSE was introduced with libgpg-error 1.21 and we now require
a later version for gnupg 2.  Thus we can switch to this more
descriptive code.

Signed-off-by: Werner Koch <wk@gnupg.org>
(cherry picked from commit f3251023750d6bd9023dbb8373c804d7d4540a56)
---
 agent/command.c | 6 +++---
 g13/server.c    | 2 +-
 sm/server.c     | 4 ++--
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/agent/command.c b/agent/command.c
index cf8a2e4..72b5973 100644
--- a/agent/command.c
+++ b/agent/command.c
@@ -2887,7 +2887,7 @@ cmd_getinfo (assuan_context_t ctx, char *line)
                 {
                   cmdopt = line;
                   if (!command_has_option (cmd, cmdopt))
-                    rc = gpg_error (GPG_ERR_GENERAL);
+                    rc = gpg_error (GPG_ERR_FALSE);
                 }
             }
         }
@@ -2901,7 +2901,7 @@ cmd_getinfo (assuan_context_t ctx, char *line)
     }
   else if (!strcmp (line, "restricted"))
     {
-      rc = ctrl->restricted? 0 : gpg_error (GPG_ERR_GENERAL);
+      rc = ctrl->restricted? 0 : gpg_error (GPG_ERR_FALSE);
     }
   else if (ctrl->restricted)
     {
@@ -2935,7 +2935,7 @@ cmd_getinfo (assuan_context_t ctx, char *line)
     }
   else if (!strcmp (line, "scd_running"))
     {
-      rc = agent_scd_check_running ()? 0 : gpg_error (GPG_ERR_GENERAL);
+      rc = agent_scd_check_running ()? 0 : gpg_error (GPG_ERR_FALSE);
     }
   else if (!strcmp (line, "std_env_names"))
     {
diff --git a/g13/server.c b/g13/server.c
index defde6c..7802952 100644
--- a/g13/server.c
+++ b/g13/server.c
@@ -530,7 +530,7 @@ cmd_getinfo (assuan_context_t ctx, char *line)
                 {
                   cmdopt = line;
                   if (!command_has_option (cmd, cmdopt))
-                    err = gpg_error (GPG_ERR_GENERAL);
+                    err = gpg_error (GPG_ERR_FALSE);
                 }
             }
         }
diff --git a/sm/server.c b/sm/server.c
index 98505e2..77ec07f 100644
--- a/sm/server.c
+++ b/sm/server.c
@@ -1162,14 +1162,14 @@ cmd_getinfo (assuan_context_t ctx, char *line)
                 {
                   cmdopt = line;
                   if (!command_has_option (cmd, cmdopt))
-                    rc = gpg_error (GPG_ERR_GENERAL);
+                    rc = gpg_error (GPG_ERR_FALSE);
                 }
             }
         }
     }
   else if (!strcmp (line, "offline"))
     {
-      rc = ctrl->offline? 0 : gpg_error (GPG_ERR_GENERAL);
+      rc = ctrl->offline? 0 : gpg_error (GPG_ERR_FALSE);
     }
   else
     rc = set_error (GPG_ERR_ASS_PARAMETER, "unknown value for WHAT");
